### PR TITLE
krb5_check_transited() ret variable is unused

### PR DESCRIPTION
**Pull request by DavidMulder**
On aix 5.3 and solaris 10 x64

transited.c: In function 'krb5_check_transited':
transited.c:607:21: error: variable 'ret' set but not used [-Werror=unused-but-set-variable]
cc1: all warnings being treated as errors

gcc 4.6.2 on both systems.

Also,

transited.c: In function '_krb5_find_capath':
transited.c:487:16: error: 'len' may be used uninitialized in this function [-Werror=uninitialized]
transited.c:450:12: note: 'len' was declared here

Same systems and gcc

Fixing commit 1501740952a815fbe10ab8345bb38cfb0ded9504

### CODE REVIEW

File: lib/krb5/transited.c [hunk 1]
```diff
@@ -447,7 +447,7 @@ hier_init(struct hier_iter *state, const char *local_realm, const char *server_r
 {
     size_t llen;
     size_t slen;
-    size_t len;
+    size_t len = 0;
```
Comment (jaltman) [inline lib/krb5/transited.c]: unrelated change?

Comment (DavidMulder) [inline lib/krb5/transited.c]: All related to Viktor Dukhovni recent changes.

File: lib/krb5/transited.c [hunk 2]
```diff
@@ -604,13 +604,12 @@ krb5_check_transited(krb5_context context,
...
 {
-    krb5_error_code ret = 0;
     char **capath = NULL;
     size_t num_capath = 0;
     size_t i = 0;
     size_t j = 0;
 
-    ret = _krb5_find_capath(context, client_realm, client_realm, server_realm,
+    _krb5_find_capath(context, client_realm, client_realm, server_realm,
                             TRUE, &capath, &num_capath);
```
Comment (jaltman) [inline lib/krb5/transited.c]: why isn't the answer to check the return code?

Comment (DavidMulder) [inline lib/krb5/transited.c]: I assumed he ignored the return code on purpose. I don't know his reasoning.
I could add a
`if (ret) return ret;`

**Comment by jaltman** [comment]
@nicowilliams Please review and fix since you merged Viktor's change.

@DavidMulder Please include the sha1 of the commit you are fixing in the commit message.

File: lib/krb5/transited.c (hunk 2 above)
Comment (vdukhovni) [inline lib/krb5/transited.c]: That's the correct fix, if properly indented.  I can commit and push that.  Or you can go ahead and update the PR.

Comment (vdukhovni) [inline lib/krb5/transited.c]: I went ahead and pushed both changes.
